src/main/java/de/dennisguse/opentracks/viewmodels/IntervalStatisticsModel.java

```diff
@@ -0,0 +1,116 @@
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import java.util.List;
+
+import de.dennisguse.opentracks.content.data.Track;
+import de.dennisguse.opentracks.content.data.TrackPoint;
+import de.dennisguse.opentracks.content.provider.ContentProviderUtils;
+import de.dennisguse.opentracks.util.PreferencesUtils;
+import de.dennisguse.opentracks.util.UnitConversions;
+
+/**
+ * This view model is used to load intervals for a track.
+ * It uses a default interval but it can be set from outside to manage the interval length.
+ */
+public class IntervalStatisticsModel extends AndroidViewModel {
+
+    private MutableLiveData<IntervalStatistics> intervalStats;
+    private IntervalOption interval;
+
+    public IntervalStatisticsModel(@NonNull Application application) {
+        super(application);
+        interval = IntervalOption.OPTION_1;
+    }
+
+    public LiveData<IntervalStatistics> getIntervalStats(Track.Id trackId, IntervalOption interval) {
+        if (intervalStats == null || this.interval != interval) {
+            intervalStats = new MutableLiveData<>();
+            this.interval = interval;
+            loadIntervalStats(trackId);
+        }
+        return intervalStats;
+    }
+
+    public LiveData<IntervalStatistics> getIntervalStats(Track.Id trackId) {
+        return getIntervalStats(trackId, interval);
+    }
+
+    /**
+     * Call this method when you want to force the view model to re-load intervals.
+     */
+    public void invalidate() {
+        intervalStats = null;
+    }
+
+    private void loadIntervalStats(Track.Id trackId) {
+        new Thread(() -> {
+            Context context = getApplication().getApplicationContext();
+            ContentProviderUtils contentProviderUtils = new ContentProviderUtils(context);
+            List<TrackPoint> trackPointList = contentProviderUtils.getTrackPoints(trackId);
+
+            IntervalStatistics intervalStatistics = new IntervalStatistics();
+            float distanceInterval = PreferencesUtils.isMetricUnits(context) ? (float) (interval.getValue() * UnitConversions.KM_TO_M) : (float) (interval.getValue() * UnitConversions.MI_TO_M);
+            intervalStatistics.build(trackPointList, distanceInterval);
+
+            intervalStats.postValue(intervalStatistics);
+        }).start();
+    }
+
+    /**
+     * Intervals length this view model support.
+     */
+    public enum IntervalOption {
+        OPTION_1(1),
+        OPTION_2(2),
+        OPTION_3(3),
+        OPTION_4(4),
+        OPTION_5(5),
+        OPTION_10(10),
+        OPTION_20(20),
+        OPTION_50(50);
+
+        private int value;
+
+        IntervalOption(int value) {
+            this.value = value;
+        }
+
+        /**
+         * @param pos position of the interval option.
+         * @return    the interval option that is in the position pos.
+         */
+        public static IntervalOption getIntervalOption(int pos) {
```

Comment:
What is the benefit of using a method?
I would also argue that we should fail immediately rather than returning a default option.

```
   IntervalOption.values();
````

Reply:
The benefit of using the method was not to fail immediately. I've removed it then ;)